### DISCUSSION TITLE
在linux中运行源码，上下课提示有几率卡住

### DISCUSSION THREAD

**Discussion by nekolfj** [Class-Widgets/Class-Widgets#204] [Q&A问答] [answered]
系统环境：archlinux 64位，pycharm2024.2.1，python3.8,3.7,3.12

在linux中运行最新源码，可能会出现上下课提示卡住（uos和archlinux，python3.7,8,12都有此问题）
在出现问题后，编译器弹出以下报错：
python3.12：
RuntimeError: wrapped C/C++ object of type ListWidget has been deleted

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/lfj/下载/Class-Widgets-main-new/Class-Widgets-main/main.py", line 1358, in update_time
    mgr.update_widgets()
  File "/home/lfj/下载/Class-Widgets-main-new/Class-Widgets-main/main.py", line 525, in update_widgets
    get_countdown(True)
  File "/home/lfj/下载/Class-Widgets-main-new/Class-Widgets-main/main.py", line 192, in get_countdown
    notification.push_notification(0, next_lessons[0])  # 下课
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lfj/下载/Class-Widgets-main-new/Class-Widgets-main/tip_toast.py", line 402, in push_notification
    main(state, lesson_name, title, subtitle, content)
  File "/home/lfj/下载/Class-Widgets-main-new/Class-Widgets-main/tip_toast.py", line 332, in main
    setTheme(Theme.AUTO)
  File "/home/lfj/下载/Class-Widgets-feat-linux-support/.venv1/lib/python3.12/site-packages/qfluentwidgets/common/style_sheet.py", line 414, in setTheme
    updateStyleSheet(lazy)
  File "/home/lfj/下载/Class-Widgets-feat-linux-support/.venv1/lib/python3.12/site-packages/qfluentwidgets/common/style_sheet.py", line 388, in updateStyleSheet
    setStyleSheet(widget, file, qconfig.theme)
  File "/home/lfj/下载/Class-Widgets-feat-linux-support/.venv1/lib/python3.12/site-packages/qfluentwidgets/common/style_sheet.py", line 326, in setStyleSheet
    widget.setStyleSheet(getStyleSheet(source, theme))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lfj/下载/Class-Widgets-feat-linux-support/.venv1/lib/python3.12/site-packages/qfluentwidgets/common/style_sheet.py", line 300, in getStyleSheet
    return applyThemeColor(source.content(theme))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lfj/下载/Class-Widgets-feat-linux-support/.venv1/lib/python3.12/site-packages/qfluentwidgets/common/style_sheet.py", line 83, in applyThemeColor
    template = QssTemplate(qss)
               ^^^^^^^^^^^^^^^^
SystemError: <class 'qfluentwidgets.common.style_sheet.QssTemplate'> returned a result with an exception set
We've got an error while stopping in post-mortem: <class 'SystemError'>

python3.8：
RuntimeError: wrapped C/C++ object of type ListWidget has been deletedThe above exception was the direct cause of the following exception:Traceback (most recent call last):File "/home/lfj/下载/Class-Widgets-main-new/Class-Widgets-main/menu.py", line 465, in <lambda>triggered=lambda: tip_toast.push_notification(3, lesson_name='信息技术')),File "/home/lfj/下载/Class-Widgets-main-new/Class-Widgets-main/tip_toast.py", line 403, in push_notificationmain(state, lesson_name, title, subtitle, content)File "/home/lfj/下载/Class-Widgets-main-new/Class-Widgets-main/tip_toast.py", line 333, in mainsetTheme(Theme.AUTO)File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 414, in setThemeupdateStyleSheet(lazy)File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 388, in updateStyleSheetsetStyleSheet(widget, file, qconfig.theme)File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 326, in setStyleSheetwidget.setStyleSheet(getStyleSheet(source, theme))File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 300, in getStyleSheetreturn applyThemeColor(source.content(theme))File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in contentreturn '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in <listcomp>return '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in contentreturn '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in <listcomp>return '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in contentreturn '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in <listcomp>return '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in contentreturn '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in <listcomp>return '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in contentreturn '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 258, in <listcomp>return '\n'.join([i.content(theme) for i in self.sources])File "/home/lfj/.local/lib/python3.8/site-packages/qfluentwidgets/common/style_sheet.py", line 217, in contentreturn self.lightStyleShee

下面视频为python3.12下debug抛出的报错：
https://www.123865.com/s/SFLBjv-Eydy3

**Comment by IsHPDuwu** [comment, marked as answer]
https://github.com/Class-Widgets/Class-Widgets/pull/569
